wsmaster/che-core-api-workspace-activity/src/main/java/org/eclipse/che/api/workspace/activity/WorkspaceActivityService.java

```diff
@@ -66,4 +75,43 @@ public void active(@ApiParam(value = "Workspace id") @PathParam("wsId") String w
       LOG.debug("Updated activity on workspace {}", wsId);
     }
   }
+
+  @GET
+  @ApiOperation("Retrieves the IDs of workspaces that have been in given state.")
+  @Produces(MediaType.APPLICATION_JSON)
+  public List<String> getWorkspacesByActivity(
```

Comment:
With adding this method class javadoc is outdated. Please update it.
Also, I think about whether this method should be in this service in a separate one, like in a special service for Metrics fetching. I'm OK to leave it as it is, just interesting to hear your opinion.

Reply:
I agree `/activity` is not the ideal REST endpoint for this kind of data. On the other hand I can't make up my mind on what's a better alternative both endpoint and code organization wise.

We have 3 options (and I can't make up my mind on which one is better):
* concentrate all the reporting endpoints under a new top level `/analytics` (or whatever) endpoint, e.g. `/analytics/workspace/`, `/analytics/user/`
    * if we also organized our codebase the same way (which seems to be the convention so far), we'd end up with a module that would be highly dependent on other modules
* keep analytical data together with the data it "analyzes". Together means in common subptath in REST as well as in the codebase to keep the inter-dependencies to the minimum, e.g. `/workspace/analytics`, `/user/analytics`
    * This keeps the analytical and "actual" code close together but fragments the analytical endpoints.
* We can also keep the code fragmented per "domain" as we do at the moment and define differently structured REST endpoints for analytics. E.g. code for workspace analytics could remain on the spot it is at the moment (ws-activity module) but define an `/analytics/workspace` REST endpoint.

Reply:
let's live it as is now. When we will see a good trend to distinct it in separate module/service we will do that. Until that moment I suggest adding```@Beta``` annotation to let people know that we not sure about the destiny of this service/method.